WooCommerce/Classes/ViewRelated/Orders/OrderDetailsViewController.swift

```diff
@@ -94,6 +94,9 @@ private extension OrderDetailsViewController {
         tableView.refreshControl = refreshControl
 
         tableView.dataSource = viewModel.dataSource
+        viewModel.dataSource.onReloadTableView = { [weak self] in
```

Comment:
`OrderDetailsViewModel` already exposes a `onUIReloadRequired` property. Would you consider setting the datasource's onReloadTableView on didSet of the OrderDetailsViewModel.onUIReloadRequired? That would make remove the need to manually setup the datasource to propagate reloads

Reply:
I just tried this, and noticed that `OrderDetailsViewModel.onUIReloadRequired`'s handler calls `OrderDetailsDataSource.reloadSections()` which triggers the async order note updates before `tableView.reloadData()`, entering an infinite loop. To avoid the infinite loop, I made this change https://github.com/woocommerce/woocommerce-ios/pull/1128/commits/657a1e1dac67a4cdd6f82ddb283dfe0f37177b66:
- Renamed `onReloadTableView` to `onUIReloadRequired` and set data source's `onUIReloadRequired` in view model's `onUIReloadRequired` `didSet`
- The responsibility of realoding data source actions is moved from `onUIReloadRequired` handler to when data source's data are set
- The `onUIReloadRequired` handler (`OrderDetailsViewController`) now only reloads table view

Lemme know if you have any other suggestions!